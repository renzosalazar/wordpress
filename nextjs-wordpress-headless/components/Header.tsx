import getMenuBySlug from '@/lib/api/queries/getMenuBySlug'

export default async function Header() {
  const menu = await getMenuBySlug('header')

  return (
    <header>
      <div>
        <h1 className="mb-0">Next.js WordPress</h1>
        <p>It&apos;s Headless WordPress</p>
      </div>
      <nav className="flex justify-between gap-4">
        {!!menu &&
          menu.menuItems.edges.map((item) => (
            <a key={item.node.databaseId} href={item.node.uri}>
              {item.node.label}
            </a>
          ))}
      </nav>
    </header>
  )
}
