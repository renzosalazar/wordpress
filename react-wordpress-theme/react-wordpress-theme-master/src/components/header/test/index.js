/**
 * Test for header component.
 *
 * @package
 */

/**
 * External dependencies
 */
import renderer from 'react-test-renderer';

/**
 * Internal dependencies
 */
import Header from '../index';

describe( 'Test Header Component', () => {
	it( 'Should match Header snapshot and render correctly', () => {
		const headerData = {
			data: {
				header: {
					siteLogoUrl: '',
					siteTitle: '',
					siteDescription: '',
					favicon: '',
					headerMenuItems: '',
				},
			},
			loading: false,
			errorMsg: '',
		};

		const component = renderer.create(
			<Header headerData={ headerData } />
		);

		const tree = component.toJSON();

		expect( tree ).toMatchSnapshot();
	} );
} );
