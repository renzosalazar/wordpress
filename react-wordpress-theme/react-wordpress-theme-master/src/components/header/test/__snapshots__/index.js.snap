// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header Component Should match Header snapshot and render correctly 1`] = `
<div
  className="header wrapper"
>
  <div
    className="logo-section"
  >
    
    <div
      className="site-info"
    >
      <h1
        className="site-title"
      >
        React WordPress Theme
      </h1>
      <p
        className="site-description"
      >
        Awesome theme
      </p>
    </div>
  </div>
  <nav
    className="header-nav menu-hidden"
  >
    
  </nav>
  <span
    className="header-nav__menu-btn"
    onClick={[Function]}
  >
    â˜°
  </span>
</div>
`;
